src/marqo/s2_inference/s2_inference.py

```diff
@@ -55,6 +54,9 @@ def vectorise(model_name: str, content: Union[str, List[str]], model_properties:
         VectoriseError: if the content can't be vectorised, for some reason.
     """
 
+    if not device:
+        raise InvalidArgError(message=f"vectorise (internal function) cannot be called without setting device!")
+    
```

Comment:
Raise a 500 error (runtimeerror) here. Based on this PR, under no circumstance, the device can be None? @pandu-k can i get your confirmation about this comment?